rest/route/host_agent.go

```diff
@@ -1085,6 +1091,30 @@ func sendBackRunningTask(ctx context.Context, env evergreen.Environment, h *host
 		return gimlet.MakeJSONInternalErrorResponder(err)
 	}
 
+	if t.IsStuckTask() {
+		if err := model.ClearAndResetStrandedHostTask(ctx, env.Settings(), h); err != nil {
+			grip.Error(message.WrapError(err, getMessage("ending and resetting system failed task")))
+			return gimlet.MakeJSONInternalErrorResponder(err)
+		}
+		// The agent is expected to run the task once it's been assigned it. If it's requesting the same
+		// task over and over again is a sign that something is wrong.
+		msg := fmt.Sprintf("The agent has re-requested the same task '%d' times.", evergreen.MaxTaskDispatchAttempts)
+		msg += " It's possible that the agent is in a bad state."
+
+		grip.Error(message.Fields{
+			"message":        msg,
+			"task_id":        t.Id,
+			"task_execution": t.Execution,
+			"host_id":        h.Id,
+			"agent_version":  h.AgentRevision,
+		})
+
+		return gimlet.MakeJSONInternalErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("The agent has re-requested the task '%s' with execution '%d' '%d' times.", t.Id, t.Execution, evergreen.MaxTaskDispatchAttempts),
```

Comment:
nit: the surrounding single quotes formatting (`'%d'`) around the number usually isn't necessary. It's only really helpful for string formatting because if the string is empty, it won't show any string at all (so it'll just look like a weird blank space).